File: src/test/java/com/thomaskint/minidao/testonly/model/ContactTest.java
package com.thomaskint.minidao.testonly.model;

import com.thomaskint.minidao.annotation.MDEntity;
import com.thomaskint.minidao.annotation.MDField;
import com.thomaskint.minidao.annotation.MDId;

import java.math.BigDecimal;

/**
 * @author Thomas Kint
 */
@MDEntity(name = "contact")
public class ContactTest {

	public static final String idContactField = "id_contact";

	@MDId
	@MDField(name = idContactField)
	public BigDecimal idContact;

	@MDField(name = "first_name")
	public String firstName;

	@MDField(name = "last_name")
	public String lastName;

	@Override
	public String toString() {
		return "ContactTest{" +
				"idContact=" + idContact +
				", firstName='" + firstName + '\'' +
				", lastName='" + lastName + '\'' +
				'}';
	}
}

File: src/test/kotlin/com/thomaskint/minidao/testonly/model/ContactTest.kt
/*
 * MIT License
 *
 * Copyright (c) 2017 Thomas Kint
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.thomaskint.minidao.testonly.model

import com.thomaskint.minidao.annotation.MDEntity
import com.thomaskint.minidao.annotation.MDField
import com.thomaskint.minidao.annotation.MDId

import java.math.BigDecimal

/**
 * @author Thomas Kint
 */
@MDEntity(tableName = "contact")
open class ContactTest(
        @MDId
        @MDField(fieldName = idContactField)
        var idContact: Long? = null,

        @MDField(fieldName = "first_name")
        var firstName: String? = null,

        @MDField(fieldName = "last_name")
        var lastName: String? = null
) {
    companion object {
        const val idContactField = "id_contact"
    }
}
